### PR TITLE
Prefer pip in documentation

### PR DESCRIPTION
**Pull request by niboshi**
We should encourage pip. Corresponding to https://github.com/chainer/chainer/pull/4366.

### CODE REVIEW

**Comment by niboshi** [comment]
@vilyaair Thanks!

File: docs/source/contribution.rst [hunk 1]
```diff
@@ -154,11 +154,11 @@ Note that we are using pytest and mock package for testing, so install them befo
 
 In order to run unit tests at the repository root, you first have to build Cython files in place by running the following command::
 
-  $ python setup.py develop
+  $ pip install .
```
Comment (kmaehashi) [inline docs/source/contribution.rst]: I think this command does not build Cython files *in place*. It should be `pip install -e .`.

Comment (niboshi) [reply docs/source/contribution.rst]: How about removing "in place" instead?
According to offline discussion with @beam2d, it seems `pip install -e .` is incompatible with `pip install .` and perhaps we should not encourage using `-e`.

Comment (niboshi) [reply docs/source/contribution.rst]: https://github.com/pypa/setuptools/issues/960

Comment (kmaehashi) [reply docs/source/contribution.rst]: I think the experiment result above is for [namespace packages](https://packaging.python.org/guides/packaging-namespace-packages/) (i.e., share the same package namespace from multiple projects.)
I'm always using `pip install -e .` and I think it is compatible. When I run `pip install -e .` then  `pip install .` (and vice versa), `pip` uninstalls the previous one and installs the new one.

Moreover, suggesting contributors to install unstable codebase (in this context s/he is currently writing a code to contribute) to global `site-packages` environment to run unit tests is not good I think.

Comment (niboshi) [reply docs/source/contribution.rst]: OK, so I think it's ok to write `-e`.

Comment (niboshi) [reply docs/source/contribution.rst]: Done.

File: docs/source/contribution.rst [hunk 2]
```diff
@@ -154,11 +154,11 @@ Note that we are using pytest and mock package for testing, so install them befo
 
 In order to run unit tests at the repository root, you first have to build Cython files in place by running the following command::
 
-  $ python setup.py develop
+  $ pip install -e .
 
 .. note::
 
-  When you modify ``*.pxd`` files, before running ``python setup.py develop``, you must clean ``*.cpp`` and ``*.so`` files once with the following command, because Cython does not automatically rebuild those files nicely::
+  When you modify ``*.pxd`` files, before running ``pip install .``, you must clean ``*.cpp`` and ``*.so`` files once with the following command, because Cython does not automatically rebuild those files nicely::
```
Comment (kmaehashi) [inline docs/source/contribution.rst]: Need to insert `-e` here too?

Comment (niboshi) [reply docs/source/contribution.rst]: Fixed.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**